sqlglot/generator.py

```diff
@@ -218,6 +222,18 @@ def sep(self, sep=" "):
     def seg(self, sql, sep=" "):
         return f"{self.sep(sep)}{sql}"
 
+    def maybe_comment(self, sql, comment, single_line=False):
+        if not comment or not self._comments:
+            return sql
+
+        if not self.pretty:
+            return f"{sql} /*{comment}*/"
```

Comment:
this is not safe, you need to do /* {comment} */ 
the space is important because you could do

select /* +hi */ 

but if you turn it into /*+ hi */, it becomes a spark hint

same in mysql with ! instead of +

Reply:
i wonder how we can do this without adding an extra space?

maybe strip out one space? haha

Reply:
We don't strip the comments anymore at all though, so if the user has `/*+ hi */` then the type hint is printed, whereas for `/* +hi */`, the comment is printed. Leading and trailing whitespace is preserved. Does that make sense?

Reply:
But for any case I'm leaving these as examples of the current behavior for reference:
```python
>>> sqlglot.transpile("SELECT /*+ BROADCAST(y) */ x.b FROM x JOIN y ON x.b = y.b", read='spark')[0]
'SELECT /*+ BROADCAST(y) */ x.b FROM x JOIN y ON x.b = y.b'
>>>
>>> sqlglot.transpile("SELECT /* +BROADCAST(y) */ x.b FROM x JOIN y ON x.b = y.b", read='spark')[0]
'SELECT /* +BROADCAST(y) */ x.b FROM x JOIN y ON x.b = y.b'
>>>
>>> sqlglot.transpile("SELECT /*+BROADCAST(y)*/ x.b FROM x JOIN y ON x.b = y.b", read='spark')[0]
'SELECT /*+ BROADCAST(y) */ x.b FROM x JOIN y ON x.b = y.b'
```

Reply:
as long as --+ doesn't turn into a hint then this is ok